core/src/main/java/org/apache/druid/indexer/TaskLocation.java

```diff
@@ -75,6 +78,19 @@ public int getTlsPort()
     return tlsPort;
   }
 
+  public URL makeURL(final String encodedPathAndQueryString) throws MalformedURLException
```

Comment:
nit: Would `makeUrl` be more camel-casey?
Although, the community as well as the Druid repo seems a little divided on this. 😃